scanner/decisions/0004-scannerv4-read-vulneralbilities.md

```diff
@@ -0,0 +1,24 @@
+# 0004 - Scanner V4 NVD CVSS enrichment data updating pipeline
+
+- **Author(s):** Yi Li
+- **Created:** [2023-09-11]
+
+## Context
+
+We are in the process of transitioning the StackRox Scanner to utilize ClairCore as its primary scanning engine. Our objective is twofold: firstly, to allow Central to fetch NVD CVSS data, and secondly, to ensure the Scanner V4 seamlessly connecting with Central, triggering data updates and enriching vulnerabilities with CVSS data. Throughout this pipeline, the Claircore enricher will play a pivotal role. It will manage tasks including downloading CVSS data from Google storage, parsing the information, and updating the data accordingly for Scanner V4
+
+## Decision
+
+The CVSS Updater resides within Central, fetching CVSS data from Google storage upon request. This updater then stores the data in Central's file system. Throughout this download and save process, the Claircore CVSS enricher plays a significant role.
+
+CVSS Updater handler and singleton in central provides http URL for sensor and scanner to connect with and send the data retrieving request. Most importantly it serves with NVD data upon those requests.
+
+In Scanner V4, the component known as the 'data retriever' (previously referred to as the 'updater' in Scanner V2). However, its name 'updater' might be confused with the 'updater' that operates as a GitHub Action workflow in the Scanner V4 context. This data retriever in Scanner V4 communicates with the CVSS data retrieval endpoint, initiating data downloads in central and ensuring access to the latest CVSS data.
```

Comment:
In the Secured Cluster, the Scanner cannot reach Central directly. The sensor needs to proxy those calls in some way.

Reply:
This CVSS data retriever in Scanner V4 matcher communicates with the CVSS data retrieval endpoint in Central.